### PR TITLE
receiver/prometheus/internal: augment metricFamilyPdata from big PR

### PR DESCRIPTION
**Pull request by odeke-em**
This change updates internal code and is meant to alleviate the
massive PR #5184 which is our eventual end-goal. No need for tests
because the code path isn't used so we have a license to update
it towards the end goal.

Particularly it:
* adds open-telemetry/opentelemetry-collector#3116 to the otlp version.
* copies the sortPoints method over verbatim.
* sets DataType and Name on pdata metrics.

Updates PR #5184

### CODE REVIEW

**Review by cuonglm — APPROVED** [review]

**Review by alolita — APPROVED** [review]

**Review by dashpole — APPROVED** [review]
Might be worth updating the description to say what this does.

It adds https://github.com/open-telemetry/opentelemetry-collector/pull/3116 to the otlp version.

It copies the sortPoints method over verbatim.

It sets type and name on pdata metrics.

**Comment by odeke-em** [comment]
> Might be worth updating the description to say what this does.
> 
> It adds [open-telemetry/opentelemetry-collector#3116](https://github.com/open-telemetry/opentelemetry-collector/pull/3116) to the otlp version.
> 
> It copies the sortPoints method over verbatim.
> 
> It sets type and name on pdata metrics.

Done, thank you @dashpole!

Thank you @alolita @dashpole @cuonglm for the reviews and approvals! Kindly cc-ing @bogdandrutu @Aneurysm9  to help with a merge.

**Review by codeboten — APPROVED** [review]

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go [hunk 1]
```diff
@@ -67,6 +69,13 @@ func newMetricFamilyPdata(metricName string, mc MetadataCache, intervalStartTime
 			metadata.Metric = familyName
 			metadata.Type = textparse.MetricTypeUnknown
 		}
+	} else if !ok && isInternalMetric(metricName) {
+		metadata = defineInternalMetric(metricName, metadata, logger)
+	}
+
+	mtype := convToPdataMetricType(metadata.Type)
+	if mtype == pdata.MetricDataTypeNone {
+		logger.Debug(fmt.Sprintf("Invalid metric : %s %+v", metricName, metadata))
```
Comment (MovieStoreGuy) [inline receiver/prometheusreceiver/internal/otlp_metricfamily.go:78]: MetricName and metadata should fields not a dynamic string to allow for easier searching in logging systems.

**Review by MovieStoreGuy — CHANGES_REQUESTED** [review]

File: receiver/prometheusreceiver/internal/otlp_metricfamily.go (hunk 1 above)
Comment (MovieStoreGuy) [reply receiver/prometheusreceiver/internal/otlp_metricfamily.go:78]: I also question the value of having the log statement here?

Comment (odeke-em) [reply receiver/prometheusreceiver/internal/otlp_metricfamily.go:78]: It's in the original code https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/0040189a514c9a28a8fb33b49ce257f4fd327320/receiver/prometheusreceiver/internal/metricfamily.go#L71-L74

this code is a port. @MovieStoreGuy I appreciate the enthusiasm but the context here is that I am porting over code, which requires careful parity changes; you can file bugs for improvements but these PRs am sending are for porting over code.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by codeboten** [comment]
@odeke-em please resolve the conflict

**Pull request closed without merging**